pandas/_libs/tslibs/period.pyx

```diff
@@ -1718,6 +1719,11 @@ cdef class _Period(PeriodMixin):
 
         return NotImplemented
 
+    def __radd__(self, other):
+        if other is NaT:
```

Comment:
this looks harmless, but is it necessary?  would falling through to `__add__` work?

Reply:
I don't think falling through works. (I vaguely remember some tests failing because of this)

Reply:
can you double-check and add a comment for when future-me thinks this looks weird and doesn't remember this thread

Reply:
Hmm. Looks like this might work? Checking with CI.